src/tls/mod.rs

```diff
@@ -168,6 +168,32 @@ pub(crate) fn tls_connector_builder(settings: &MaybeTlsSettings) -> Result<SslCo
     let mut builder = SslConnector::builder(SslMethod::tls()).context(TlsBuildConnectorSnafu)?;
     if let Some(settings) = settings.tls() {
         settings.apply_context(&mut builder)?;
+        let mut min_proto_version = SslVersion::TLS1;
```

Comment:
Repetitive, could you extract a re-usable function or method returning min_version + ciphersuites? Pardon my Rust, I don't know what that would look like.

Reply:
@alanconway Please take another look.